### ISSUE TITLE
Crayfish problem with opening netcdf4 in Windows

### ISSUE THREAD

**Issue by saberraz** [lutraconsulting/qgis-crayfish-plugin#257]
It works under Linux with gdal 2.1.3 but fails in Windows with gdal 2.1.2

**Comment by saberraz** [comment]
Same problem with SHOM data (data.shom.fr).

Example files are available for testing.

**Comment by PeterPetrik** [comment]
http://gis.stackexchange.com/questions/179132/gdal-nc-rasters-opened-as-netcdf-network-common-data-format-under-linux-an

**Comment by PeterPetrik** [comment]
@wonder-sk can you help me with this one please? my findings so far:

1) it works on windows 32 bit and linux 
2) on windows 64 it throws strange errors 

```
ERROR 1: netCDF scanline fetch failed: #-101 (NetCDF: HDF error)
ERROR 1: NETCDF:"H:\GIS\crayfish\netCDF\NC4\WFLOWoutmapsNC4.nc":iflow, band 365: IReadBlock failed at X offset 0, Y offset 0
ERROR 1: GetBlockRef failed at X block offset 0, Y block offset 0
````

3) the nc4 works for test case qgis-crayfish-plugin\test\data\NetCDF\indonesia_nc4.nc, but failing for others. 
4) the error is the same I got for other grib file from
 https://github.com/lutraconsulting/qgis-crayfish-plugin/issues/267
This file is possible to open in previous crayfish release.

I think that it can
a) relate to some change in gdal reader code (both formats are read by GDAL) in 2.5.0 (reopening gdal dataset twice?)
b) or/and some problem in libraries used by gdal (netcdf and grib)?

**Comment by PeterPetrik** [comment]
https://trac.osgeo.org/osgeo4w/ticket/293

the file is not opened correctly on win64 in QGIS too (all times 0, values NaN)

**Comment by PeterPetrik** [comment]
even when using OSGeo4W64b to do ncdump, it errors out 

`H:\GIS\crayfish\netCDF\NC4>ncdump WFLOWoutmapsNC4.nc > WFLOWoutmapsNC4.txt
NetCDF: HDF error
Location: file ..\..\..\ncdump\vardata.c; line 475`

**Comment by saberraz** [comment]
@arthurvd, this is an upstream bug with the OSGeo4W installer with Windows 64-bit.

It works as expected in Linux and Windows 32-bit. We propose to write the results to netcdf 3 until this bug is resolved in QGIS installer.

**Comment by saberraz** [comment]
This has been fixed upstream in gdal/ogr 2.20. Checked the problematic nc files and Crayfish can handle them fine under Windows too,
